### DISCUSSION TITLE
Thank you

### DISCUSSION THREAD

**Discussion by sombriks** [sombriks/node-libgpiod#6] [General]
I've enabled the discussion on this project and would like starting this with a great THANK YOU!

@noctarius and @dmitrydvorkin i thank you so much for your contributions, because the repo name will draw some attention and at my current work i am not having enough free time to polish it properly.

so assuming libgpiod will indeed remain as major way to modern linux distros will handle gpio stuff, would be a pleasure to me to have you as first class contributors, able to merge things and take important design decisions.

what do you think? 

i understand that me alone myself can't make this wrapper a first class product, mas you guys seems to know better than me and honestly i would appreciate so much if you accept this invitation to accept the chance to call the shots there with me and make node-libgpiod something bigger than i alone could make it to be.

**Comment by noctarius** [comment]
Hey @sombriks, thanks for the kind offer, I appreciate it. Since there isn't a ton of feature / pull requests here, I'm happy to step in and help where possible, especially since we use the library ourselves and I'm interested in keeping it alive :-)

**Comment by dmitrydvorkin** [comment]
No problem, @sombriks !

I am interested in this library too, case there are the only proper way to do GPIO - is to use modern Linux GPIO interface.
Everything else is the piece of shit or looks like the trash bin.

One thing I'd like to note: it is not really necessary to use libgpiod library it is just the abstraction on top of the very simple mechanism that could be used directly. Probably I'll make an attempt to get rid of this dependency.

It could also be useful to implement Event-driven IN. I have a wish to do that, but it depends on my free time. Event-driven input brings a lot of optimization. It should definitely be done as separate object (just like polling -in is done).

I am already including your library in our company distribution: https://github.com/tibbotech/yocto_layers

**Comment by sombriks** [comment]
awesome to hear @dmitrydvorkin !

the event-driven IN is also an 'old' wish of mine, it's in the roadmap of the project and hits the free time wall too.

but it's there and someday we'll get it!
